src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -937,34 +914,33 @@ int
 SHAMap::unshare()
 {
     // Don't share nodes wth parent map
-    return walkSubTree(false, hotUNKNOWN, 0);
+    return walkSubTree(false, hotUNKNOWN);
 }
 
-/** Convert all modified nodes to shared nodes */
-// If requested, write them to the node store
 int
-SHAMap::flushDirty(NodeObjectType t, std::uint32_t seq)
+SHAMap::flushDirty(NodeObjectType t)
 {
-    return walkSubTree(true, t, seq);
+    // We only write back if this map is backed.
+    return walkSubTree(backed_, t);
 }
 
 int
-SHAMap::walkSubTree(bool doWrite, NodeObjectType t, std::uint32_t seq)
+SHAMap::walkSubTree(bool doWrite, NodeObjectType t)
```

Comment:
This function has a new precondition: if `doWrite` is true, then `backed_` must be true as well. Maybe add an assert?
```
assert(!doWrite || backed_);
```